include/CXXGraph/Node/Node_impl.hpp

```diff
@@ -70,6 +70,18 @@ void Node<T>::setData(T &&new_data) {
   this->data = std::move(new_data);
 }
 
+ bool deleteNodeById(const std::string &id) {
+    auto it = std::find_if(nodes.begin(), nodes.end(), [&](const auto &pair) {
+      return pair.second.getUserId() == id;
+    });
```

Comment:
Need to include `algorithm` in order to use `std::find_if`

Reply:
YES